From 60bb09c10292ff6d78b984dd0f3d36fdda4aa768 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Wed, 4 Oct 2023 14:30:36 +0200
Subject: [PATCH 131/162] fix(LinstorSR): restart drbd-reactor in case of
 failure

Otherwise we can have all hosts unusable after a massive reboot:
all the drbd-reactor services are marked as failed, and no controller
is started.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 Makefile                                                | 3 +++
 etc/systemd/system/drbd-reactor.service.d/override.conf | 6 ++++++
 2 files changed, 9 insertions(+)
 create mode 100644 etc/systemd/system/drbd-reactor.service.d/override.conf

diff --git a/Makefile b/Makefile
index f8196cb..42cdfa9 100755
--- a/Makefile
+++ b/Makefile
@@ -148,6 +148,7 @@ install: precheck
 	mkdir -p $(SM_STAGING)$(UDEV_SCRIPTS_DIR)
 	mkdir -p $(SM_STAGING)$(INIT_DIR)
 	mkdir -p $(SM_STAGING)$(SYSTEMD_CONF_DIR)
+	mkdir -p $(SM_STAGING)$(SYSTEMD_CONF_DIR)/drbd-reactor.service.d
 	mkdir -p $(SM_STAGING)$(SYSTEMD_CONF_DIR)/linstor-satellite.service.d
 	mkdir -p $(SM_STAGING)$(SYSTEMD_SERVICE_DIR)
 	mkdir -p $(SM_STAGING)$(MPATH_CONF_DIR)
@@ -177,6 +178,8 @@ install: precheck
 	  $(SM_STAGING)/$(SM_DEST)
 	install -m 644 etc/logrotate.d/$(SMLOG_CONF) \
 	  $(SM_STAGING)/$(LOGROTATE_DIR)
+	install -m 644 etc/systemd/system/drbd-reactor.service.d/override.conf \
+	  $(SM_STAGING)/$(SYSTEMD_CONF_DIR)/drbd-reactor.service.d/
 	install -m 644 etc/systemd/system/linstor-satellite.service.d/override.conf \
 	  $(SM_STAGING)/$(SYSTEMD_CONF_DIR)/linstor-satellite.service.d/
 	install -m 644 etc/systemd/system/var-lib-linstor.service \
diff --git a/etc/systemd/system/drbd-reactor.service.d/override.conf b/etc/systemd/system/drbd-reactor.service.d/override.conf
new file mode 100644
index 0000000..2f99a46
--- /dev/null
+++ b/etc/systemd/system/drbd-reactor.service.d/override.conf
@@ -0,0 +1,6 @@
+[Service]
+StartLimitInterval=60
+StartLimitBurst=10
+
+Restart=always
+RestartSec=2
-- 
2.43.0

